terraform/provider.tf

```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "~> 3.0,!= 3.14.0"
+      version = "~> 5.0"
```

Comment:
Wouldn't it be better to pin to something like `>3.0 and <6`? Since we're not actually using any features exclusive to the newer provider I don't see why we would require using a newer version if it will work just as well with an older one. I'm not as familiar with terraform providers as I am with python dependencies, but this is what I would do for a "library" such as this in a python dependency tree.

Reply:
Then you also wouldn't have to bump the major version of the release because you wouldn't be potentially breaking anything on projects using an older provider.

Reply:
From terraform docs: https://developer.hashicorp.com/terraform/tutorials/configuration-language/provider-versioning

> Terraform will download the latest provider version that fulfills the version constraint.

So I think setting the version bounds here as wide as possible to all known compatible providers is the best move.

Reply:
OK, actually the official guidance looks to recommend not even setting an upper limit: https://developer.hashicorp.com/terraform/language/expressions/version-constraints#terraform-core-and-provider-versions

> Reusable modules should constrain only their minimum allowed versions of Terraform and providers, such as `>= 0.12.0`. This helps avoid known incompatibilities, while allowing the user of the module flexibility to upgrade to newer versions of Terraform without altering the module.